File: Classification_Algorithms/simpleLinearRegression.py
# -*- coding: utf-8 -*-
"""
Created on Fri Oct 13 14:51:22 2017

@author: bpras
"""
#index in python start at 0 -----------------------------------------------
#importing libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importing data set
dataset = pd.read_csv('Salary_Data.csv')
#get all rows and columns except last column
X = dataset.iloc[:, :-1].values
#get only the third colum
Y = dataset.iloc[:, 1].values

#splitting the dataset into training and test sets
from sklearn.cross_validation import train_test_split
X_train, X_test, Y_train , Y_test = train_test_split(X, Y, test_size = 1/3, random_state = 0)

#Feature scaling not necessary here.. algorithm will take care of it here
#fitting the simple linear regression to the training set
from sklearn.linear_model import LinearRegression
regressor = LinearRegression()
regressor.fit(X_train,Y_train)

#Predicting the test set results 
Y_pred = regressor.predict(X_test)

#visualize training and test set results 
plt.scatter(X_train, Y_train, color = 'red') # training points Xtrain, Ytrain
plt.plot(X_train, regressor.predict(X_train), color = 'blue') # regression line Xtrain, YtrainPrediction
plt.title('Salary Vs Experience (Training set)')
plt.xlabel('Years of experience')
plt.ylabel('Salary')
plt.show()

plt.scatter(X_test, Y_test, color = 'green') # training points Xtrain, Ytrain
plt.plot(X_train, regressor.predict(X_train), color = 'blue') # regression line Xtrain, YtrainPrediction
# changing above line to Xtest vs YtestPrediction, results remains same.. because our regressor is aleady
#trained on the training set. So linear regression model remains the same
plt.title('Salary Vs Experience (Test set)')
plt.xlabel('Years of experience')
plt.ylabel('Salary')
plt.show()

File: Classification_Algorithms/simpleLinearRegression.R
#author - Prashant Bhat


#import dataset 
#index starts with '1'
dataset = read.csv('Salary_Data.csv')

#split dataset into train and test
#install.packages('caTools')
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
#if true , its in training set.. 0.8 percent of entries will be in training set
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

#fitting data to regressor - liner model 
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

#predicting test results 
y_pred = predict(regressor, newdata = test_set)

#visualize the results 
#install.packages('ggplot2')
library(ggplot2)
ggplot()+
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary), 
             colour = 'red')+
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'blue')+
  ggtitle('Salary vs Exerience (Training set)')+
  xlab('Years of Experience')+
  ylab('Salary')

ggplot()+
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary), 
             colour = 'red')+
  geom_line(aes(x = test_set$YearsExperience, y = predict(regressor, newdata = test_set)),
            colour = 'green')+
  ggtitle('Salary vs Exerience (Test set)')+
  xlab('Years of Experience')+
  ylab('Salary')











